src/main/java/net/mcreator/blockly/BlocklyToCode.java

```diff
@@ -88,10 +91,13 @@ public final String getGeneratedCode() {
 
 	public final String getExtraTemplatesCode() throws TemplateGeneratorException {
 		StringBuilder code = new StringBuilder();
-		if (templateGenerator != null) {
-			for (String template : usedTemplates) {
+		while (!usedTemplates.isEmpty() && templateGenerator != null) {

```

Comment:
This code makes it look like templateGenerator can become null and we thus watch this in while.

Delegate templateGenerator null check above while loop, either wrap it, or return prematurely with empty string